java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -160,25 +164,30 @@ protected void selectVerticalAutoTickUnit(Graphics2D g2, Rectangle2D dataArea, R
         TickUnit unit1 = tickUnits.getCeilingTickUnit(this.getTickUnit());
 
         //Smallest tick distance could be at either end
-        final double unitHeightUpper = Math.abs(this.lengthToJava2D(getRange().getUpperBound(), dataArea, edge) - this.lengthToJava2D(getRange().getUpperBound() - unit1.getSize(), dataArea, edge));
-        final double unitHeightLower = Math.abs(this.lengthToJava2D(getRange().getLowerBound(), dataArea, edge) - this.lengthToJava2D(getRange().getLowerBound() + unit1.getSize(), dataArea, edge));
+        final double unitHeightUpper = Math.abs(this.lengthToJava2D(Math.abs(getRange().getUpperBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getUpperBound() - unit1.getSize()),
+                dataArea,
+                edge));
+        final double unitHeightLower = Math.abs(this.lengthToJava2D(Math.abs(getRange().getLowerBound()), dataArea, edge) - this.lengthToJava2D(Math.abs(getRange().getLowerBound() + unit1.getSize()),
```

Comment:
Is it expensive to call getRange() multiple times?  Should it just be called once and used multiple times?

Reply:
Not really expensive, but since it does some computation I would say it should be cached

Reply:
cached